hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/RemoteProcedureDispatcher.java

```diff
@@ -370,39 +383,47 @@ public TRemote getKey() {
     }
 
     @Override
-    public synchronized void add(final RemoteProcedure operation) {
-      if (this.operations == null) {
```

Comment:
Are you sure we are matching this condition with `timeoutExecutor.putIfAbsent(this);`?
Based on `putIfAbsent()` method, it seems only first time when entry is added for new key in map i.e when concurrent map's thread-safe version of `putIfAbsent()` returns null, we execute this block but is there any scenario that might not adhere to this `if (this.operations == null) ` condition by any chance?

Reply:
TimeoutExecutorThread#run will remove task from pendingBufferNode， so after run， task will not be included in the  pendingBufferNode
else {
     pendingBufferNode.remove(task);
     ((BufferNode) task).dispatch();
 }